system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -1,28 +1,56 @@
 const React = require("react");
 const {connect} = require("react-redux");
+const classNames = require("classnames");
+const LinkMenu = require("content-src/components/LinkMenu/LinkMenu");
 
-function displayURL(url) {
-  return new URL(url).hostname.replace(/^www./, "");
+class TopSites extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {showContextMenu: false, activeTile: null};
+    this.displayURL = this.displayURL.bind(this);
+  }
+  displayURL(url) {
+    return new URL(url).hostname.replace(/^www./, "");
+  }
+  toggleContextMenu(event, index) {
+    this.setState({showContextMenu: true, activeTile: index});
+  }
+  render() {
+    return (<section>
+      <h3 className="section-title">Top Sites</h3>
+      <ul className="top-sites-list">
+        {this.props.TopSites.rows.map((link, index) => {
+          const isActive = this.state.showContextMenu && this.state.activeTile === index;
+          const title = this.displayURL(link.url);
+          const className = `screenshot${link.screenshot ? " active" : ""}`;
+          const style = {backgroundImage: (link.screenshot ? `url(${link.screenshot})` : "none")};
+          return (<li key={link.url}>
+            <div className={classNames("tile-outer", {active: isActive})}>
```

Comment:
Can't you just use the `li`? Or is there a specific reason you need this extra div (also could you rename this to `top-site-wrapper` or `top-site-outer` to make it more descriptive)